mongomock/aggregate.py

```diff
@@ -634,6 +635,62 @@ def _handle_date_operator(self, operator, values):
                     ' in Mongomock.'
                 )
             return out_value['date'].strftime(out_value['format'])
+        if operator == '$dateFromParts':
+            if not isinstance(values, dict):
+                raise OperationFailure(
+                    '$dateFromParts operator must correspond a dict '
+                    'that has "year" or "isoWeekYear" field.'
+                )
+            if (
+                not isinstance(values, dict)
+                or len(set(values.keys()).intersection({'year', 'isoWeekYear'})) != 1
+            ):
+                raise OperationFailure(
+                    f'{operator} operator must correspond a dict '
+                    'that has "year" or "isoWeekYear" field.'
+                )
+            if 'isoWeekYear' in values.keys():
+                raise NotImplementedError(
+                    'Although isoWeekYear is a valid field for the '
+                    f'{operator} operator, it is currently not implemented '
+                    ' in Mongomock.'
+                )
+            if 'isoWeek' in values.keys():
+                raise NotImplementedError(
+                    'Although isoWeek is a valid field for the '
+                    f'{operator} operator, it is currently not implemented '
+                    ' in Mongomock.'
+                )
+            if 'isoDayOfWeek' in values.keys():
+                raise NotImplementedError(
+                    'Although isoDayOfWeek is a valid field for the '
+                    f'{operator} operator, it is currently not implemented '
+                    ' in Mongomock.'
+                )
+            if 'timezone' in values.keys():
+                raise NotImplementedError(
+                    'Although timezone is a valid field for the '
+                    f'{operator} operator, it is currently not implemented '
+                    ' in Mongomock.'
+                )
+
+            year = out_value.get('year')
```

Comment:
For all the checks above you have used `values` instead of `out_value` to get the data. I prefer that you keep using the same (`$dateToString` is a bad example here).